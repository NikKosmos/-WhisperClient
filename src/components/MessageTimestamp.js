import React from 'react'
import PropTypes from 'prop-types'

function MessageTimestamp(props) {

    const date = new Date(props.time)
    const hours = date.getHours()
    const minutes = date.getMinutes()
    const day = date.getDate()
    const month = date.getMonth() + 1
    const year = date.getFullYear()

    return (
        <div>
            <p>{ hours }:{ minutes } { day }.{ month }.{ year }</p>
        </div>
    )
}

MessageTimestamp.propTypes = {
    time: PropTypes.oneOfType([PropTypes.number, PropTypes.string, PropTypes.instanceOf(Date)]).isRequired
}

export { MessageTimestamp }