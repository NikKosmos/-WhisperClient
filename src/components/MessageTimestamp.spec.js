import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json';

import { MessageTimestamp } from './MessageTimestamp'

describe('NewMessageInput', () => {
    const mockTime = "2019-12-25T17:02:10.480Z";

    it ('should render correctly', () => {
        const testedComponent = shallow(<MessageTimestamp 
            time = { mockTime }
        />)

        expect(shallowToJson(testedComponent)).toMatchSnapshot();
      })

})