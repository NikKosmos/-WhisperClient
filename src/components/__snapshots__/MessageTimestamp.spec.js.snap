// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewMessageInput should render correctly 1`] = `
<div>
  <p>
    21
    :
    2
     
    25
    .
    12
    .
    2019
  </p>
</div>
`;
